packages/blast-verify/verify.js

```diff
@@ -0,0 +1,200 @@
+
+    // Submit a job to verify the contract
+    const form = new FormData()
+    form.append('address', contractAddress)
+    form.append('crateName', localContractLabel)
+    form.append('optimizer', `cosmwasm/workspace-optimizer:${getRustOptimizerVersion()}`)
+    form.append('source', fs.createReadStream(outputArchiveDir))
+    console.log('Uploading contracts for verification...')
+    verificationResponse = await axios.post(`${apiUrl}/verify-contract`, form, { headers: { ...form.getHeaders() } })
+  // eslint-disable-next-line no-useless-catch
+  } catch (err) {
+    throw err
+  } finally {
+    // Always delete the archive
+    fs.rmSync(tempDir, {
+      recursive: true, force: true
+    })
+  }
+  if (verificationResponse.status !== 201) {
+    throw new BlastVerifyError(`Verify contract API request failed! Response status: ${verificationResponse.status}`)
+  }
+
+  const id = verificationResponse.data.id
+  console.log('Verification job is submitted. Getting the result may take a long time. Now you can opt to safely free' +
+    ' your terminal by killing the process (ctrl+C) and check the status manually on ' +
+    `${apiUrl}/verification-status?id=${id}`)
+
+  // Periodically check if the verification job is done
+  console.log('Verifying contract...')
+  let isVerified = false
+  while (1) {
+    // Pause the process for 5 seconds
+    await new Promise((resolve) => {
+      setTimeout(resolve, 5 * 1000)
+    })
+    const { data } = await axios.get(`${apiUrl}/verification-status?id=${id}`)
+
+    // checks for error messages from API
+    if (data.verificationError && data.verificationError.length > 0) {
+      console.log(`Verification failed! Reason: ${data.verificationError}`)
+      return data
+    }
+    if (data.parsingError && data.parsingError.length > 0) {
+      console.log(`Parsing failed! Reason: ${data.parsingError}`)
+      return data
+    }
+    // watch for the progress depending on API response
+    if (!isVerified && data.verified === true) {
+      isVerified = true
+      console.log('Smart contract is verified successfully! Parsing schema...')
+    }
+    if (data.parsed === true) {
+      console.log('Schema is parsed successfully!')
+      return data
+    }
+  }
+}
+
+const getVerifyUrlFromNetwork = () => {
+  const { config } = getConfig()
+
+  if (!config.verify) {
+    throw new BlastVerifyError('Missing [verify] from the config file.')
+  }
+  if (!config.verify.network) {
+    throw new BlastVerifyError('Missing [verify.network] from the config file.')
+  }
+  if (!networks[config.verify.network]) {
+    throw new BlastVerifyError('Invalid network passed in the config file! Use "blast verify --ls" to show ' +
+      'all available networks.')
+  }
+  return networks[config.verify.network]
+}
+
+const createContractsArchive = (outputArchiveDir) => {
+  const contractsDir = path.join(getProjectRootPath(), 'contracts')
+  const packagesDir = path.join(getProjectRootPath(), 'packages')
```

Comment:
why aren't you checking for packagesDIr?

Reply:
because in case of contracts that do not import anything, `packages` folder is not required